src/main/java/com/impactupgrade/nucleus/client/SfdcClient.java

```diff
@@ -611,6 +611,31 @@ protected QueryResult querySmsContacts(String updatedSinceClause, String filter,
     return query(query);
   }
 
+  public List<QueryResult> getDonorContacts(Calendar updatedSince, String... extraFields)
+      throws ConnectionException, InterruptedException {
+    List<QueryResult> queryResults = new ArrayList<>();
+
+    String updatedSinceClause = "";
+    if (updatedSince != null) {
+      updatedSinceClause = "SystemModStamp >= " + new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'").format(updatedSince.getTime());
+    }
+    queryResults.add(queryDonorContacts(updatedSinceClause, extraFields));
+
+    return queryResults;
+  }
+
+  protected QueryResult queryDonorContacts(String updatedSinceClause, String... extraFields) throws ConnectionException, InterruptedException {
```

Comment:
@VSydor One thing I realized we're missing in this: querying for business accounts. Check out AccountingService.getDonationContact. Similar to what we're doing for the MC sync, if the donation only had an Account associated with it and not a Contact, we might still need to create a faux CrmContact. Otherwise, that organization will not have created a Xero Contact by the time we sync its donations.

Ex: https://destinyrescue-fnsf.lightning.force.com/lightning/r/Opportunity/006J3000003wRMNIA2/view (business gift, no contact on the gift, no primary contact defined on the account).

So we might have to introduce that faux contact concept for this as well (further upstream, though)? Query for all non-household Accounts that are donors and updated since the date, create faux CrmContacts for them, and sync those to Xero as well?

Actually, I partially take that back. Since they want the "ContactPerson" included, we have to get a contact in addition to the business? Query for non-household accounts, get their primary contacts, and then return those as CrmContacts with the CrmAccount included inside it?

Not all Accounts have primary contacts defined on them in SFDC. Many do, but it's optional. In the doc, I have a convo going in the doc with Maria on what to do in this case. Most of them WILL at least have an affiliation defined that we could use to get a Contact...